numba_dppy/driver/usm_ndarray_type.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from numba.extending import typeof_impl, register_model
+from numba_dppy.dppy_array_type import DPPYArray, DPPYArrayModel
+import numba_dppy.target as dppy_target
+from dpctl.tensor import usm_ndarray
+from numba.np import numpy_support
+
+
+class USM_NdArrayType(DPPYArray):
```

Comment:
docstring needed.

Reply:
Also, please use something like https://github.com/IntelPython/dpctl/blob/a826e04c454b3c41965c5ae8edda94aa89946d5d/dpctl/memory/_memory.pyx#L596 so that the generated docs show the whole signature for the class constructor.